### PR TITLE
Moving to Python3.x on OsX

### PR DESCRIPTION
**Pull request by lekshmideepu**
At the moment OsX build uses Python2.x. This PR will fix one of the task in #1313 and  #1325

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -68,8 +68,8 @@ jobs:
       env: MATRIX_EVAL="CC=clang-7 && CXX=clang++-7" xRUN_BUILD_AND_TESTSUITE=1
     - stage: OsX
       os: osx
-      osx_image: xcode10.2
-      language: generic
+      osx_image: xcode10.3 # Python 3.7.4 running on macOS 10.14.4
+      language: shell # 'language: python' is an error on Travis CI macOS
```
Comment (jougs) [inline .travis.yml]:
```suggestion
      language: shell      # languages 'generic' or 'python' result in errors on macOS
```

File: .travis.yml [hunk 2]
```diff
@@ -150,6 +150,8 @@ before_install:
          if [ "$TRAVIS_OS_NAME" = "osx" ]; then
            brew --version
            brew update
+           brew info python
+           export PATH="/usr/local/opt/python/libexec/bin:$PATH" #To use homebrew's python, add the directory in the PATH.
```
Comment (jougs) [inline .travis.yml]:
```suggestion
           export PATH="/usr/local/opt/python/libexec/bin:$PATH"  # To use homebrew's Python
```

File: .travis.yml (hunk 1 above)
Comment (lekshmideepu) [reply .travis.yml]: @jougs Thanks for reviewing.  `language: generic` works in general for OsX. Our current setup uses it.

**Comment by lekshmideepu** [comment]
> @lekshmideepu I approve, conditional on changes by Jochen being accepted.

Thanks for the review @heplesser

File: .travis.yml [hunk 3]
```diff
@@ -68,8 +68,8 @@ jobs:
       env: MATRIX_EVAL="CC=clang-7 && CXX=clang++-7" xRUN_BUILD_AND_TESTSUITE=1
     - stage: OsX
       os: osx
-      osx_image: xcode10.2
-      language: generic
+      osx_image: xcode10.3 # Python 3.7.4 running on macOS 10.14.4
+      language: shell # language 'python' result in errors on macOS
```
Comment (jougs) [inline .travis.yml]:
```suggestion
      language: shell      # language 'python' results in errors on macOS
```

**Comment by jougs** [comment]
I'll fix the grammar after the merge.

**Comment by lekshmideepu** [comment]
> I'll fix the grammar after the merge.

Thanks :-)

**Pull request merged**